### ISSUE TITLE
Ansi tests: &key arguments not correctly checked

### ISSUE THREAD

**Issue by kpoeck** [clasp-developers/clasp#461]
Example
````
CL-USER> (defun blah (a &key foo bar)
	   (list a foo bar))
BLAH
CL-USER> (blah 1 :foo)
(1 0 NIL)
````
Expected a PROGRAM-ERROR complaining a missing value for the key

**Comment by Bike** [comment]
The problem is probably in cmp/arguments.lsp. But it's a horrible mess

**Comment by Bike** [comment]
Should be fixed by 589e9319770c4bc1f13cd9c2e23b81868bd1416f

**Comment by kpoeck** [comment]
works!, great job!

That should solve a lot of ansi-tests, will count later and report.
